test/e2e/storage/csimock/csi_attach_volume.go

```diff
@@ -109,6 +110,66 @@ var _ = utils.SIGDescribe("CSI Mock volume attach", func() {
 		}
 	})
 
+	ginkgo.Context("When CSIDriver AttachRequired changes from true to false", func() {
+		f.It("should cleanup VolumeAttachment properly", f.WithSlow(), func(ctx context.Context) {
+			var err error
+			m.init(ctx, testParameters{registerDriver: true, disableAttach: false})
+			ginkgo.DeferCleanup(m.cleanup)
+
+			driverName := m.config.GetUniqueDriverName()
+			csiDriver, err := m.cs.StorageV1().CSIDrivers().Get(ctx, driverName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Failed to get CSIDriver: %v", err)
+			driver := csiDriver.DeepCopy()
+
+			_, claim, pod := m.createPod(ctx, pvcReference)
+			if pod == nil {
+				return
+			}
+			testConfig := storageframework.ConvertTestConfig(m.config)
+			attachmentName := e2evolume.GetVolumeAttachmentName(ctx, m.cs, testConfig, m.provisioner, claim.Name, claim.Namespace)
+			ginkgo.DeferCleanup(func(ctx context.Context) {
+				_, err := m.cs.StorageV1().VolumeAttachments().Get(ctx, attachmentName, metav1.GetOptions{})
+				if err == nil {
+					err := m.cs.StorageV1().VolumeAttachments().Delete(ctx, attachmentName, metav1.DeleteOptions{})
+					framework.ExpectNoError(err, "Failed to delete VolumeAttachment: %v", err)
+				} else if !apierrors.IsNotFound(err) {
+					framework.ExpectNoError(err, "Failed to get VolumeAttachment")
+				}
+			})
+
```

Comment:
It makes code simpler.

```suggestion
		    ginkgo.DeferCleanup(framework.IgnoreNotFound(m.cs.StorageV1().VolumeAttachments().Delete), attachmentName, metav1.DeleteOptions{})
```

Reply:
updated.